<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
  <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
  <meta http-equiv="Content-Style-Type" content="text/css" />
  <meta name="generator" content="pandoc" />
  <title></title>
  <style type="text/css">code{white-space: pre;}</style>
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <!-- jQuery -->
  <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/2.0.3/jquery.min.js"></script>
  <script src="https://cdnjs.cloudflare.com/ajax/libs/jqueryui/1.10.3/jquery-ui.min.js"></script>
  
  <!-- bootstrap -->
  <!--<link href="https://netdna.bootstrapcdn.com/bootstrap/3.0.0/css/bootstrap.min.css" rel="stylesheet"  id="style">-->
  <script src="https://netdna.bootstrapcdn.com/bootstrap/3.0.0/js/bootstrap.min.js"></script>
  
  <!-- highlight.js -->
  <!--<link href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/7.3/styles/default.min.css" rel="stylesheet" id="code-style">-->
  <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/7.3/highlight.min.js"></script>
  <script>
  hljs.LANGUAGES.r=function(a){var b="([a-zA-Z]|\\.[a-zA-Z.])[a-zA-Z0-9._]*";return{c:[a.HCM,{b:b,l:b,k:{keyword:"function if in break next repeat else for return switch while try tryCatch|10 stop warning require library attach detach source setMethod setGeneric setGroupGeneric setClass ...|10",literal:"NULL NA TRUE FALSE T F Inf NaN NA_integer_|10 NA_real_|10 NA_character_|10 NA_complex_|10"},r:0},{cN:"number",b:"0[xX][0-9a-fA-F]+[Li]?\\b",r:0},{cN:"number",b:"\\d+(?:[eE][+\\-]?\\d*)?L\\b",r:0},{cN:"number",b:"\\d+\\.(?!\\d)(?:i\\b)?",r:0},{cN:"number",b:"\\d+(?:\\.\\d*)?(?:[eE][+\\-]?\\d*)?i?\\b",r:0},{cN:"number",b:"\\.\\d+(?:[eE][+\\-]?\\d*)?i?\\b",r:0},{b:"`",e:"`",r:0},{cN:"string",b:'"',e:'"',c:[a.BE],r:0},{cN:"string",b:"'",e:"'",c:[a.BE],r:0}]}}(hljs); </script>
  <!--<script type="text/javascript", src="https://yandex.st/highlightjs/7.3/languages/r.min.js"></script>-->
  
  <!-- Manific Popup -->
  <script src="https://cdnjs.cloudflare.com/ajax/libs/magnific-popup.js/0.8.9/jquery.magnific-popup.min.js"></script>
  
  <script type="text/javascript"
    src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML">
  </script>
  <script defer="defer">
  // Function to generate the dynamic table of contents
  jQuery.fn.generate_TOC = function () {
    var base = $(this[0]);
  
    var selectors = ['h1', 'h2', 'h3', 'h4'];
  
    var last_ptr = [{}, {}, {}, {}];
  
    var anchors = {};
  
    generate_anchor = function (text) {
      var test = text.replace(/\W/g, '_');
  
      while(test in anchors){
        //if no suffix, add one
        if(test.match(/_\d+$/) === null){
          test = test + "_2";
        }
        //else generate unique id for duplicates by adding one to the suffix
        else {
          test = test.replace(/_(\d+)$/, function(match, number){ var num=+number+1; return("_" + num) });
        }
      }
      anchors[test]=1;
      return(test);
    }
  
    $(selectors.join(',')).filter(function(index) { return $(this).parent().attr("id") != 'header'; }).each(function () {
  
      var heading = $(this);
      var idx = selectors.indexOf(heading.prop('tagName').toLowerCase());
      var itr = 0;
  
      while (itr <= idx) {
        if (jQuery.isEmptyObject(last_ptr[itr])) {
          last_ptr[itr] = $('<ul>').addClass('nav');
          if (itr === 0) {
            base.append(last_ptr[itr])
          } else {
            if(last_ptr[itr-1].children('li').length === 0){
              last_ptr[itr-1].append(last_ptr[itr]);
            }
            else {
              last_ptr[itr - 1].children('li').last().append(last_ptr[itr]);
            }
          }
        }
        itr++;
      }
      var anchor = generate_anchor(heading.text());
      heading.attr('id', anchor);
      var a = $('<a>')
      .text(heading.text())
      .attr('href', '#' + anchor);
  
    var li = $('<li>')
      .append(a);
  
    last_ptr[idx].append(li);
    for (i = idx + 1; i < last_ptr.length; i++) {
      last_ptr[i] = {};
    }
    });
  }
  /* run scripts when document is ready */
  $(function() {
    "use strict";
  
    var $window = $(window);
    var $body = $(document.body);
  
    /* size of thumbnails */
  
    var hidden_types = ['source']
    var output_types = ['output', 'message', 'warning', 'error']
  
    /* style tables */
    $('table').addClass('table table-striped table-bordered table-hover table-condensed');
  
    $('pre code').each(function(i, e) {
      hljs.highlightBlock(e);
    });
  
    /* Magnific Popup */
    $(".thumbnail").each(function(){
      $(this).magnificPopup({
        disableOn: 768,
        closeOnContentClick: true,
  
        type: 'image',
        items: {
          src: $(this).find('img').attr('src'),
        }
      });
    });
  
    function toggle_block(obj, show) {
      var span = obj.find('span');
      if(show === true){
        span.removeClass('glyphicon-chevron-up').addClass('glyphicon-chevron-down');
        obj.next('pre').slideDown();
      }
      else {
        span.removeClass('glyphicon-chevron-down').addClass('glyphicon-chevron-up');
        obj.next('pre').slideUp();
      }
    }
  
    function toggle_thumbnails(imgs, show){
      if(show === true){
        imgs.parents().show()
        imgs.slideDown();
      }
      else {
        imgs.slideUp(400, function(){ $(this).parent().hide(); });
      }
    }
  
    function global_toggle(obj){
      var type = obj.attr('type');
      var show = !obj.parent('li').hasClass('active');
      if(show === true){
        obj.parent('li').addClass('active');
      }
      else{
        obj.parent('li').removeClass('active');
      }
      if(type == 'figure'){
        toggle_thumbnails($('.thumbnail img'), show);
      }
      else {
        $('.toggle.' + type).each(function() { toggle_block($(this), show); });
      }
    }
  
    /* onclick toggle next code block */
    $('.toggle').click(function() {
      var span = $(this).find('span');
      toggle_block($(this), !span.hasClass('glyphicon-chevron-down'));
      return false
    })
  
    // global toggles
    $('.toggle-global').click(function(){
      var type = $(this).attr('type');
      if(type === 'all-source'){
          $('li a.source').each(function() {
            global_toggle($(this));
          });
        }
      else if(type === 'all-output'){
        $.each(output_types, function(i, val){
          console.log(val);
          global_toggle($('li a.' + val));
        });
      }
      else {
        console.log($(this));
        global_toggle($(this));
      }
      return false;
    });
    /* table of contents */
    if($(['h1', 'h2', 'h3', 'h4'].join(',')).length > 0){
      $('body > #wrap > .container > .row').append('<div class="col-md-2"><div id="toc" class="well sidebar sidenav affix hidden-print"/></div>');
      $('#toc').generate_TOC();
    }
  
    $.each(hidden_types, function(i, type) {
      $('li[type=' + type + ']').each(function(){ global_toggle($(this)); });
    });
  
    /* remove paragraphs with no content */
    $('p:empty').remove();
  
    $body.scrollspy({
      target: '.sidebar',
    });
  
    /* theme switch */
    $('.theme-switch').click(function(){
      var css = $('link[title=' + $(this).attr('title') + ']');
      $('#theme[rel=stylesheet]').attr('href', css.attr('href'));
      $('.theme-switch').closest('li').removeClass('active');
      $(this).closest('li').addClass('active');
      return false;
    });
    /* code style switch */ //TODO use same function for both of these?
    $('.highlight-switch').click(function(){
      var css = $('link[title="' + $(this).attr('title') + '"]');
      $('#highlight[rel=stylesheet]').attr('href', css.attr('href'));
      $('.highlight-switch').closest('li').removeClass('active');
      $(this).closest('li').addClass('active');
      return false;
    });
  
    //TODO refresh on show/hide
    $window.on('load', function () {
      $body.scrollspy('refresh');
    })
  
  });
  
  </script>
  <style>
  /* Knitr_bootstrap styles */
  #header {
    display: none !important;
    visibility: hidden !important;
  }
  #wrap .container-fluid {
    padding: 0;
    overflow: hidden;
  }
  .toggle{
    text-transform: capitalize;
  }
  
  .toggle-global{
    text-transform: capitalize;
  }
  
  /* Sticky footer styles */
  * {
    margin:0;
  }
  html,
  body {
      height: 100%;
      padding:0 !important;
      /* The html and body elements cannot have any padding or margin. */
      /*overflow-x: hidden;*/
  }
  
  /* Wrapper for page content to push down footer */
  #wrap {
      min-height: 100%;
      height: auto !important;
      height: 100%;
      /* Negative indent footer by it's height */
      margin: 0 auto -120px;
  }
  
  /* Set the fixed height of the footer here */
  #push,
  #footer {
      height: 120px;
  }
  
  #footer {
    text-align: center;
  }
  
  /* Top level subheader elements.  These are the first nested items underneath a header element. */
  .header li {
    font-size: 20px;
  }
  
  /* Makes the font smaller for all subheader elements. */
  .sub-header li {
      font-size: 12px;
  }
  
  button.thumbnails {
    margin-left:0px;
  }
  
  /*
   * Side navigation
   *
   * Scrollspy and affixed enhanced navigation to highlight sections and secondary
   * sections of docs content.
   */
  
  /* By default it's not affixed in mobile views, so undo that */
  .sidebar.affix {
    position: static;
  }
  
  /* First level of nav */
  .sidenav {
    margin-top: 30px;
    margin-bottom: 30px;
    padding-top:    10px;
    padding-bottom: 10px;
    border-radius: 5px;
  }
  
  /* All levels of nav */
  .sidebar .nav > li > a {
    display: block;
    padding: 5px 20px;
  }
  .sidebar .nav > li > a:hover,
  .sidebar .nav > li > a:focus {
    text-decoration: none;
    border-right: 1px solid;
  }
  .sidebar .nav > .active > a,
  .sidebar .nav > .active:hover > a,
  .sidebar .nav > .active:focus > a {
    font-weight: bold;
    background-color: transparent;
    border-right: 1px solid;
  }
  
  /* Nav: second level (shown on .active) */
  .sidebar .nav .nav {
    display: none; /* Hide by default, but at >768px, show it */
    margin-bottom: 8px;
  }
  .sidebar .nav .nav > li > a {
    padding-top:    3px;
    padding-bottom: 3px;
    padding-left: 30px;
    font-size: 90%;
  }
  
  .sidebar .nav .nav .nav > li > a {
    padding-left: 40px;
  }
  .sidebar .nav .nav .nav .nav > li > a {
    padding-left: 50px;
  }
  
  /* Show and affix the side nav when space allows it */
  @media screen and (min-width: 992px) {
    .sidebar .nav > .active > ul {
      display: block;
    }
    /* Widen the fixed sidebar */
    .sidebar.affix,
    .sidebar.affix-bottom {
      width: 213px;
    }
    .sidebar.affix-top,
    .sidebar.affix {
      position: fixed; /* Undo the static from mobile first approach */
      top: 30px;
    }
    .sidebar.affix-bottom {
      position: absolute; /* Undo the static from mobile first approach */
    }
    .sidebar.affix-bottom .sidenav,
    .sidebar.affix .sidenav {
      margin-top: 0;
      margin-bottom: 0;
    }
  }
  @media screen and (min-width: 1200px) {
    /* Widen the fixed sidebar again */
    .sidebar.affix-bottom,
    .sidebar.affix {
      width: 263px;
    }
  }
  
  #toc {
    padding: 10px 0px;
    margin:0;
    border:0;
  }
  
  
  .panel pre {
    margin: 0;
    padding: 0;
    border: 0;
  }
  button + pre {
    margin: 0;
    padding: 0;
  }
  pre code {
    border-radius: 0;
  }
  /* Magnific Popup CSS */
  .mfp-bg {
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    z-index: 1042;
    overflow: hidden;
    position: fixed;
    background: #0b0b0b;
    opacity: 0.8;
    filter: alpha(opacity=80); }
  
  .mfp-wrap {
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    z-index: 1043;
    position: fixed;
    outline: none !important;
    -webkit-backface-visibility: hidden; }
  
  .mfp-container {
    text-align: center;
    position: absolute;
    width: 100%;
    height: 100%;
    left: 0;
    top: 0;
    padding: 0 8px;
    -webkit-box-sizing: border-box;
    -moz-box-sizing: border-box;
    box-sizing: border-box; }
  
  .mfp-container:before {
    content: '';
    display: inline-block;
    height: 100%;
    vertical-align: middle; }
  
  .mfp-align-top .mfp-container:before {
    display: none; }
  
  .mfp-content {
    position: relative;
    display: inline-block;
    vertical-align: middle;
    margin: 0 auto;
    text-align: left;
    z-index: 1045; }
  
  .mfp-inline-holder .mfp-content,
  .mfp-ajax-holder .mfp-content {
    width: 100%;
    cursor: auto; }
  
  .mfp-ajax-cur {
    cursor: progress; }
  
  .mfp-zoom-out-cur,
  .mfp-zoom-out-cur .mfp-image-holder .mfp-close {
    cursor: -moz-zoom-out;
    cursor: -webkit-zoom-out;
    cursor: zoom-out; }
  
  .mfp-zoom {
    cursor: pointer;
    cursor: -webkit-zoom-in;
    cursor: -moz-zoom-in;
    cursor: zoom-in; }
  
  .mfp-auto-cursor .mfp-content {
    cursor: auto; }
  
  .mfp-close,
  .mfp-arrow,
  .mfp-preloader,
  .mfp-counter {
    -webkit-user-select: none;
    -moz-user-select: none;
    user-select: none; }
  
  .mfp-loading.mfp-figure {
    display: none; }
  
  .mfp-hide {
    display: none !important; }
  
  .mfp-preloader {
    color: #cccccc;
    position: absolute;
    top: 50%;
    width: auto;
    text-align: center;
    margin-top: -0.8em;
    left: 8px;
    right: 8px;
    z-index: 1044; }
  
  .mfp-preloader a {
    color: #cccccc; }
  
  .mfp-preloader a:hover {
    color: white; }
  
  .mfp-s-ready .mfp-preloader {
    display: none; }
  
  .mfp-s-error .mfp-content {
    display: none; }
  
  button.mfp-close,
  button.mfp-arrow {
    overflow: visible;
    cursor: pointer;
    background: transparent;
    border: 0;
    -webkit-appearance: none;
    display: block;
    padding: 0;
    z-index: 1046;
    -webkit-box-shadow: none;
    box-shadow: none; }
  
  button::-moz-focus-inner {
    padding: 0;
    border: 0; }
  
  .mfp-close {
    width: 44px;
    height: 44px;
    line-height: 44px;
    position: absolute;
    right: 0;
    top: 0;
    text-decoration: none;
    text-align: center;
    opacity: 0.65;
    padding: 0 0 18px 10px;
    color: white;
    font-style: normal;
    font-size: 28px;
    font-family: Arial, Baskerville, monospace; }
    .mfp-close:hover, .mfp-close:focus {
      opacity: 1; }
    .mfp-close:active {
      top: 1px; }
  
  .mfp-close-btn-in .mfp-close {
    color: #333333; }
  
  .mfp-image-holder .mfp-close,
  .mfp-iframe-holder .mfp-close {
    color: white;
    right: -6px;
    text-align: right;
    padding-right: 6px;
    width: 100%; }
  
  .mfp-counter {
    position: absolute;
    top: 0;
    right: 0;
    color: #cccccc;
    font-size: 12px;
    line-height: 18px; }
  
  .mfp-arrow {
    position: absolute;
    opacity: 0.65;
    margin: 0;
    top: 50%;
    margin-top: -55px;
    padding: 0;
    width: 90px;
    height: 110px;
    -webkit-tap-highlight-color: rgba(0, 0, 0, 0); }
  
  .mfp-arrow:active {
    margin-top: -54px; }
  
  .mfp-arrow:hover,
  .mfp-arrow:focus {
    opacity: 1; }
  
  .mfp-arrow:before, .mfp-arrow:after,
  .mfp-arrow .mfp-b,
  .mfp-arrow .mfp-a {
    content: '';
    display: block;
    width: 0;
    height: 0;
    position: absolute;
    left: 0;
    top: 0;
    margin-top: 35px;
    margin-left: 35px;
    border: medium inset transparent; }
  .mfp-arrow:after,
  .mfp-arrow .mfp-a {
    border-top-width: 13px;
    border-bottom-width: 13px;
    top: 8px; }
  .mfp-arrow:before,
  .mfp-arrow .mfp-b {
    border-top-width: 21px;
    border-bottom-width: 21px; }
  
  .mfp-arrow-left {
    left: 0; }
    .mfp-arrow-left:after,
    .mfp-arrow-left .mfp-a {
      border-right: 17px solid white;
      margin-left: 31px; }
    .mfp-arrow-left:before,
    .mfp-arrow-left .mfp-b {
      margin-left: 25px;
      border-right: 27px solid #3f3f3f; }
  
  .mfp-arrow-right {
    right: 0; }
    .mfp-arrow-right:after,
    .mfp-arrow-right .mfp-a {
      border-left: 17px solid white;
      margin-left: 39px; }
    .mfp-arrow-right:before,
    .mfp-arrow-right .mfp-b {
      border-left: 27px solid #3f3f3f; }
  
  .mfp-iframe-holder {
    padding-top: 40px;
    padding-bottom: 40px; }
  
  .mfp-iframe-holder .mfp-content {
    line-height: 0;
    width: 100%;
    max-width: 900px; }
  
  .mfp-iframe-scaler {
    width: 100%;
    height: 0;
    overflow: hidden;
    padding-top: 56.25%; }
  
  .mfp-iframe-scaler iframe {
    position: absolute;
    display: block;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    box-shadow: 0 0 8px rgba(0, 0, 0, 0.6);
    background: black; }
  
  .mfp-iframe-holder .mfp-close {
    top: -40px; }
  
  /* Main image in popup */
  img.mfp-img {
    width: auto;
    max-width: 100%;
    height: auto;
    display: block;
    line-height: 0;
    -webkit-box-sizing: border-box;
    -moz-box-sizing: border-box;
    box-sizing: border-box;
    padding: 40px 0 40px;
    margin: 0 auto; }
  
  /* The shadow behind the image */
  .mfp-figure:after {
    content: '';
    position: absolute;
    left: 0;
    top: 40px;
    bottom: 40px;
    display: block;
    right: 0;
    width: auto;
    height: auto;
    z-index: -1;
    box-shadow: 0 0 8px rgba(0, 0, 0, 0.6);
    background: #444444; }
  
  .mfp-figure {
    line-height: 0; }
  
  .mfp-bottom-bar {
    margin-top: -36px;
    position: absolute;
    top: 100%;
    left: 0;
    width: 100%;
    cursor: auto; }
  
  .mfp-title {
    text-align: left;
    line-height: 18px;
    color: #f3f3f3;
    word-wrap: break-word;
    padding-right: 36px; }
  
  .mfp-figure small {
    color: #bdbdbd;
    display: block;
    font-size: 12px;
    line-height: 14px; }
  
  .mfp-image-holder .mfp-content {
    max-width: 100%; }
  
  .mfp-gallery .mfp-image-holder .mfp-figure {
    cursor: pointer; }
  
  @media screen and (max-width: 800px) and (orientation: landscape), screen and (max-height: 300px) {
    /**
     * Remove all paddings around the image on small screen
     */
    .mfp-img-mobile .mfp-image-holder {
      padding-left: 0;
      padding-right: 0; }
  
    .mfp-img-mobile img.mfp-img {
      padding: 0; }
  
    /* The shadow behind the image */
    .mfp-img-mobile .mfp-figure:after {
      top: 0;
      bottom: 0; }
  
    .mfp-img-mobile .mfp-bottom-bar {
      background: rgba(0, 0, 0, 0.6);
      bottom: 0;
      margin: 0;
      top: auto;
      padding: 3px 5px;
      position: fixed;
      -webkit-box-sizing: border-box;
      -moz-box-sizing: border-box;
      box-sizing: border-box; }
  
    .mfp-img-mobile .mfp-bottom-bar:empty {
      padding: 0; }
  
    .mfp-img-mobile .mfp-counter {
      right: 5px;
      top: 3px; }
  
    .mfp-img-mobile .mfp-close {
      top: 0;
      right: 0;
      width: 35px;
      height: 35px;
      line-height: 35px;
      background: rgba(0, 0, 0, 0.6);
      position: fixed;
      text-align: center;
      padding: 0; }
  
    .mfp-img-mobile .mfp-figure small {
      display: inline;
      margin-left: 5px; } }
  @media all and (max-width: 900px) {
    .mfp-arrow {
      -webkit-transform: scale(0.75);
      transform: scale(0.75); }
  
    .mfp-arrow-left {
      -webkit-transform-origin: 0;
      transform-origin: 0; }
  
    .mfp-arrow-right {
      -webkit-transform-origin: 100%;
      transform-origin: 100%; }
  
    .mfp-container {
      padding-left: 6px;
      padding-right: 6px; } }
  .mfp-ie7 .mfp-img {
    padding: 0; }
  .mfp-ie7 .mfp-bottom-bar {
    width: 600px;
    left: 50%;
    margin-left: -300px;
    margin-top: 5px;
    padding-bottom: 5px; }
  .mfp-ie7 .mfp-container {
    padding: 0; }
  .mfp-ie7 .mfp-content {
    padding-top: 44px; }
  .mfp-ie7 .mfp-close {
    top: 0;
    right: 0;
    padding-top: 0; }
  
  //Magnific overrides
  .mfp-image img{
    background: white;
  }
  .mfp-figure:after {
    background: white;
  }
  
  /*
   * Off Canvas navbar toggle right
   * --------------------------------------------------
   */
  
  @media screen and (max-width: 768px) {
    .row-offcanvas .collapsing {
    -webkit-transition: none 0;
      -moz-transition: none 0;
      transition: none 0;
    }
   .row-offcanvas .navbar {
    position: absolute;
    z-index: 2;
      right:0;
      height:100%;
      width:55px;
      border:0;
      background-color:transparent;
    }
    .row-offcanvas .navbar-toggle {
      margin-right: 5px;
      margin-left: 5px;
    }
    .row-offcanvas {
      position: relative;
    }
    .row-offcanvas-right.active .navbar {
    position: absolute;
    z-index: 2;
      right: -28.4%;
      width:40%;
      background-color:#eee;
      border:0 solid #ddd;
      border-left-width:1px;
    }
    .row-offcanvas-right.active {
      left: -30%;
    }
    .row-offcanvas-right.active .navbar-collapse {
      position: relative;
      width: 100%;
    }
    .row-offcanvas .content {
    /*width:calc(100% - 60px);*/
    }
  }
  </style>
</head>
<body>
<div id="wrap">
<div class="container">
<div class="row row-offcanvas row-offcanvas-right">
<div class="contents col-xs-12 col-md-10">
<!--
%\VignetteEngine{knitr::rmarkdown}
%\VignetteIndexEntry{Introduction to regionReport}
-->
<h1>HTML reports for a set of regions</h1>
<p>If you wish, you can view this vignette online <a href="http://lcolladotor.github.io/regionReport/">here</a>.</p>
<p><code>regionReport</code> <a id='cite-regionReport'></a>(<a href='https://github.com/lcolladotor/regionReport'>Collado-Torres, Jaffe, and Leek, 2014</a>) creates HTML reports styled with <code>knitrBootstrap</code> <a id='cite-knitrBootstrap'></a>(<a href='https://github.com/jimhester/'>Hester, 2014</a>) for a set of regions such as <code>derfinder</code> <a id='cite-derfinder'></a>(<a href='https://github.com/lcolladotor/derfinder'>Collado-Torres, Frazee, Jaffe, and Leek, 2014</a>) results.</p>
<p>Currently, this package includes a basic exploration analysis of <code>derfinder</code> results which we expect users to be interested in reproducing with their own data. The analysis is written in <a href="http://www.rstudio.com/ide/docs/r_markdown">R Markdown</a> format and <code>derfinderReport()</code> takes the results from <code>derfinder</code>, performs a couple of setup operations, and then relies on <code>knitr</code> <a id='cite-knitr'></a>(<a href='http://www.crcpress.com/product/isbn/9781466561595'>Xie, 2014</a>), <code>rmarkdown</code> <a id='cite-rmarkdown'></a>(<a href='http://CRAN.R-project.org/package=rmarkdown'>Allaire, McPherson, Xie, Wickham, et al., 2014</a>), and <code>knitrBootstrap</code> (<a href='https://github.com/jimhester/'>Hester, 2014</a>) for generating the report.</p>
<h1>Using <code>regionReport</code></h1>
<h2>General case</h2>
<p>This is currently under construction.</p>
<h2><code>derfinder</code> case</h2>
<p>Goodies in this report are powered by <code>ggbio</code> <a id='cite-ggbio'></a>(<a href='#bib-ggbio'>Yin, Cook, and Lawrence, 2012</a>) and <code>ggplot2</code> <a id='cite-ggplot2'></a>(<a href='http://had.co.nz/ggplot2/book'>Wickham, 2009</a>).</p>
<h3>Run <code>derfinder</code></h3>
<p>Prior to using <code>regionReport::derfinderReport()</code> you must use <code>derfinder</code> to analyze a specific data set. While there are many ways to do so, we recommend using <strong>analyzeChr()</strong> with the same <em>prefix</em> argument. Then merging the results with <strong>mergeResults()</strong>.</p>
<p>Below, we run <code>derfinder</code> for the example data included in the package. The steps are:</p>
<ol style="list-style-type: decimal">
<li>Load derfinder</li>
<li>Create a directory where we'll store the results</li>
<li>Generate the pre-requisites for the models to use with the example data</li>
<li>Generate the statistical models</li>
<li>Analyze the example data for chr21</li>
<li>Merge the results (only one chr in this case, but in practice there'll be more)</li>
</ol>
<div class="row">
<button class="source R toggle btn btn-xs btn-primary">
<span class="glyphicon glyphicon-chevron-down"></span> R source
</button>
<pre style=""><code class="source r">## Load derfinder
library('derfinder')

## The output will be saved in the 'report' directory
dir.create('report', showWarnings = FALSE, recursive = TRUE)</code></pre>
</div>
<p>The following code runs <code>derfinder</code>.</p>
<div class="row">
<button class="source R toggle btn btn-xs btn-primary">
<span class="glyphicon glyphicon-chevron-down"></span> R source
</button>
<pre style=""><code class="source r">## Save the current path
initialPath <- getwd()
setwd(file.path(initialPath, 'report'))

## Generate output from derfinder

## Collapse the coverage information
collapsedFull <- collapseFullCoverage(list(genomeData$coverage), 
verbose=TRUE)

## Calculate library size adjustments
sampleDepths <- sampleDepth(collapsedFull, probs=c(0.5), nonzero=TRUE, 
verbose=TRUE)

## Build the models
group <- genomeInfo$pop
adjustvars <- data.frame(genomeInfo$gender)
models <- makeModels(sampleDepths, testvars=group, adjustvars=adjustvars)

## Analyze chromosome 21
analysis <- analyzeChr(chr='21', coverageInfo=genomeData, models=models, 
cutoffFstat=1, cutoffType='manual', seeds=20140330, groupInfo=group, 
mc.cores=1, writeOutput=TRUE, returnOutput=TRUE)

## Save the stats options for later
optionsStats <- analysis$optionsStats

## Change the directory back to the original one
setwd(initialPath)</code></pre>
</div>
<p>For convenience, we have included the <code>derfinder</code> results as part of <code>regionReport</code>. Note that the above functions are routinely checked as part of <code>derfinder</code>.</p>
<div class="row">
<button class="source R toggle btn btn-xs btn-primary">
<span class="glyphicon glyphicon-chevron-down"></span> R source
</button>
<pre style=""><code class="source r">## Copy previous results
file.copy(system.file(file.path('extdata', 'chr21'), package='derfinder', 
mustWork=TRUE), 'report', recursive=TRUE)</code></pre>
<button class="output R toggle btn btn-xs btn-success">
<span class="glyphicon glyphicon-chevron-down"></span> R output
</button>
<pre style=""><code class="output r">## [1] TRUE
</code></pre>
</div>
<p>Next, proceed to merging the results.</p>
<div class="row">
<button class="source R toggle btn btn-xs btn-primary">
<span class="glyphicon glyphicon-chevron-down"></span> R source
</button>
<pre style=""><code class="source r">## Merge the results from the different chromosomes. In this case, there's 
## only one: chr21
mergeResults(chrs = 'chr21', prefix = 'report',
    genomicState = genomicState$fullGenome)</code></pre>
<button class="message R toggle btn btn-xs btn-info">
<span class="glyphicon glyphicon-chevron-up"></span> R message
</button>
<pre style="display:none"><code class="message r">## 2014-09-29 22:32:43 mergeResults: Saving options used
## 2014-09-29 22:32:43 Loading chromosome chr21
## Loading required package: parallel
## 
## Attaching package: 'BiocGenerics'
## 
## The following objects are masked from 'package:parallel':
## 
##     clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,
##     clusterExport, clusterMap, parApply, parCapply, parLapply,
##     parLapplyLB, parRapply, parSapply, parSapplyLB
## 
## The following object is masked from 'package:stats':
## 
##     xtabs
## 
## The following objects are masked from 'package:base':
## 
##     anyDuplicated, append, as.data.frame, as.vector, cbind,
##     colnames, do.call, duplicated, eval, evalq, Filter, Find, get,
##     intersect, is.unsorted, lapply, Map, mapply, match, mget,
##     order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,
##     rbind, Reduce, rep.int, rownames, sapply, setdiff, sort,
##     table, tapply, union, unique, unlist, unsplit
## 
## Loading required package: stats4
## Neither 'cutoffFstatUsed' nor 'optionsStats' were supplied, so the FWER calculation step will be skipped.
## 2014-09-29 22:32:44 mergeResults: Saving fullNullSummary
## 2014-09-29 22:32:44 mergeResults: Re-calculating the p-values
## 2014-09-29 22:32:44 mergeResults: Saving fullRegions
## 2014-09-29 22:32:44 mergeResults: assigning genomic states
## 2014-09-29 22:32:44 annotateRegions: counting
## 2014-09-29 22:32:44 annotateRegions: annotating
## 2014-09-29 22:32:44 mergeResults: Saving fullAnnotatedRegions
## 2014-09-29 22:32:44 mergeResults: Saving fullFstats
## 2014-09-29 22:32:44 mergeResults: Saving fullTime
</code></pre>
</div>
<h3>Create report</h3>
<p>Once the <code>derfinder</code> output has been generated and merged, use <strong>derfinderReport()</strong> to create the HTML report.</p>
<div class="row">
<button class="source R toggle btn btn-xs btn-primary">
<span class="glyphicon glyphicon-chevron-down"></span> R source
</button>
<pre style=""><code class="source r">## Load derfindeReport
library('regionReport')</code></pre>
</div>
<div class="row">
<button class="source R toggle btn btn-xs btn-primary">
<span class="glyphicon glyphicon-chevron-down"></span> R source
</button>
<pre style=""><code class="source r">## Generate the HTML report
report <- derfinderReport(prefix='report', browse=FALSE,
    nBestRegions=15, makeBestClusters=TRUE, outdir='html',
    fullCov=list('21'=genomeDataRaw$coverage), optionsStats=optionsStats)</code></pre>
</div>
<p>Once the output is generated, you can browse the report from <code>R</code> using <strong>browseURL()</strong> as shown below.</p>
<div class="row">
<button class="source R toggle btn btn-xs btn-primary">
<span class="glyphicon glyphicon-chevron-down"></span> R source
</button>
<pre style=""><code class="source r">## Browse the report
browseURL(report)</code></pre>
</div>
<p>You can compare the resulting report with the pre-compiled report using the following code.</p>
<div class="row">
<button class="source R toggle btn btn-xs btn-primary">
<span class="glyphicon glyphicon-chevron-down"></span> R source
</button>
<pre style=""><code class="source r">browseURL(system.file(file.path('basicExploration', 'basicExploration.html'),  
    package = 'regionReport', mustWork = TRUE))</code></pre>
</div>
<h3>Notes</h3>
<p>Note that the reports require an active Internet connection to render correctly.</p>
<p>The report is self-explanatory and will change some of the text depending on the input options.</p>
<p>If the report is taking too long to compile (say more than 3 hours), you might want to consider setting <em>nBestCluters</em> to a small number or even set <em>makeBestClusters</em> to <code>FALSE</code>.</p>
<h1>Advanced arguments</h1>
<p>If you are interested in using the advanced arguments, use <code>derfinder::advancedArg()</code> as shown below:</p>
<div class="row">
<button class="source R toggle btn btn-xs btn-primary">
<span class="glyphicon glyphicon-chevron-down"></span> R source
</button>
<pre style=""><code class="source r">## URLs to advanced arguemtns
derfinder::advancedArg('derfinderReport', package = 'regionReport',
    browse = FALSE)
## Set browse = TRUE if you want to open them in your browser</code></pre>
</div>
<h1>Reproducibility</h1>
<p>This package was made possible thanks to:</p>
<ul>
<li>R <a id='cite-R'></a>(<a href='http://www.R-project.org/'>R Core Team, 2014</a>)</li>
<li>derfinder (<a href='https://github.com/lcolladotor/derfinder'>Collado-Torres, Frazee, Jaffe, and Leek, 2014</a>)</li>
<li>derfinderPlot <a id='cite-derfinderPlot'></a>(<a href='https://github.com/lcolladotor/derfinderPlot'>Collado-Torres, Jaffe, and Leek, 2014</a>)</li>
<li>devtools <a id='cite-devtools'></a>(<a href='http://CRAN.R-project.org/package=devtools'>Wickham and Chang, 2014</a>)</li>
<li>GenomeInfoDb <a id='cite-GenomeInfoDb'></a>(<a href='#bib-GenomeInfoDb'>Arora, Morgan, Carlson, and Pages, 2014</a>)</li>
<li>GenomicRanges <a id='cite-GenomicRanges'></a>(<a href='http://www.ploscompbiol.org/article/info%3Adoi%2F10.1371%2Fjournal.pcbi.1003118'>Lawrence, Huber, Pagès, Aboyoun, et al., 2013</a>)</li>
<li>ggbio (<a href='#bib-ggbio'>Yin, Cook, and Lawrence, 2012</a>)</li>
<li>ggplot2 (<a href='http://had.co.nz/ggplot2/book'>Wickham, 2009</a>)</li>
<li>grid (<a href='http://www.R-project.org/'>R Core Team, 2014</a>)</li>
<li>gridExtra <a id='cite-gridExtra'></a>(<a href='http://CRAN.R-project.org/package=gridExtra'>Auguie, 2012</a>)</li>
<li>IRanges (<a href='http://www.ploscompbiol.org/article/info%3Adoi%2F10.1371%2Fjournal.pcbi.1003118'>Lawrence, Huber, Pagès, Aboyoun, et al., 2013</a>)</li>
<li>knitcitations <a id='cite-knitcitations'></a>(<a href='https://github.com/cboettig/knitcitations'>Boettiger, 2014</a>)</li>
<li>knitr (<a href='http://www.crcpress.com/product/isbn/9781466561595'>Xie, 2014</a>)</li>
<li>knitrBootstrap (<a href='https://github.com/jimhester/'>Hester, 2014</a>)</li>
<li>mgcv</li>
<li>RColorBrewer <a id='cite-RColorBrewer'></a>(<a href='http://CRAN.R-project.org/package=RColorBrewer'>Neuwirth, 2011</a>)</li>
<li>rmarkdown (<a href='http://CRAN.R-project.org/package=rmarkdown'>Allaire, McPherson, Xie, Wickham, et al., 2014</a>)</li>
<li>biovizBase <a id='cite-biovizBase'></a>(<a href='#bib-biovizBase'>Yin, Lawrence, and Cook, 2014</a>)</li>
<li>Cairo <a id='cite-Cairo'></a>(<a href='http://CRAN.R-project.org/package=Cairo'>Urbanek and Horner, 2013</a>)</li>
<li>TxDb.Hsapiens.UCSC.hg19.knownGene <a id='cite-TxDbHsapiensUCSChg19knownGene'></a>(<a href='#bib-TxDbHsapiensUCSChg19knownGene'>Carlson, 2014</a>)</li>
</ul>
<p>Code for creating the vignette</p>
<div class="row">
<button class="source R toggle btn btn-xs btn-primary">
<span class="glyphicon glyphicon-chevron-down"></span> R source
</button>
<pre style=""><code class="source r">## Create the vignette
library('knitrBootstrap') 

knitrBootstrapFlag <- packageVersion('knitrBootstrap') < '1.0.0'
if(knitrBootstrapFlag) {
    ## CRAN version
    system.time(knit_bootstrap('regionReport.Rmd', chooser=c('boot', 'code'), show_code = TRUE))
    unlink('regionReport.md')
} else {
    ## GitHub version
    library('rmarkdown')
    system.time(render('regionReport.Rmd',
        'knitrBootstrap::bootstrap_document'))
}
## Note: if you prefer the knitr version use:
# library('rmarkdown')
# system.time(render('regionReport.Rmd', 'html_document'))

## Extract the R code
library('knitr')
knit('regionReport.Rmd', tangle = TRUE)

## Copy report output to be distributed with the package for comparison 
## purposes
if(gsub('.*/', '', getwd()) == 'realVignettes') {
    file.copy(file.path('report', 'html', 'basicExploration.html'),
        file.path('..', '..', 'inst', 'basicExploration',
            'basicExploration.html'), overwrite=TRUE)
} else {
    file.copy(file.path('report', 'html', 'basicExploration.html'),
        file.path('..', 'inst', 'basicExploration', 'basicExploration.html'),
            overwrite=TRUE)
}

       
## Clean up
file.remove('regionReportRef.bib')
#unlink('regionReport_files', recursive=TRUE)
unlink('report', recursive = TRUE)</code></pre>
</div>
<p>Date the vignette was generated.</p>
<div class="row">
<button class="output R toggle btn btn-xs btn-success">
<span class="glyphicon glyphicon-chevron-down"></span> R output
</button>
<pre style=""><code class="output r">## [1] "2014-09-29 22:33:59 EDT"
</code></pre>
</div>
<p>Wallclock time spent generating the vignette.</p>
<div class="row">
<button class="output R toggle btn btn-xs btn-success">
<span class="glyphicon glyphicon-chevron-down"></span> R output
</button>
<pre style=""><code class="output r">## Time difference of 1.504 mins
</code></pre>
</div>
<p><code>R</code> session information.</p>
<div class="row">
<button class="message R toggle btn btn-xs btn-info">
<span class="glyphicon glyphicon-chevron-down"></span> R message
</button>
<pre style=""><code class="message r">## Session info--------------------------------------------------------------
</code></pre>
<button class="output R toggle btn btn-xs btn-success">
<span class="glyphicon glyphicon-chevron-down"></span> R output
</button>
<pre style=""><code class="output r">##  setting  value                       
##  version  R version 3.1.1 (2014-07-10)
##  system   x86_64, darwin10.8.0        
##  ui       X11                         
##  language (EN)                        
##  collate  en_US.UTF-8                 
##  tz       America/New_York
</code></pre>
<button class="message R toggle btn btn-xs btn-info">
<span class="glyphicon glyphicon-chevron-down"></span> R message
</button>
<pre style=""><code class="message r">## Packages------------------------------------------------------------------
</code></pre>
<button class="output R toggle btn btn-xs btn-success">
<span class="glyphicon glyphicon-chevron-down"></span> R output
</button>
<pre style=""><code class="output r">##  package           * version  date      
##  acepack             1.3.3.3  2013-05-03
##  AnnotationDbi       1.27.15  2014-09-24
##  base64enc           0.1.2    2014-06-26
##  BatchJobs           1.3      2014-07-10
##  BBmisc              1.7      2014-06-21
##  bibtex              0.3.6    2013-07-29
##  Biobase             2.25.0   2014-04-12
##  BiocGenerics      * 0.11.5   2014-09-13
##  BiocParallel        0.99.22  2014-09-24
##  biomaRt             2.21.1   2014-07-08
##  Biostrings          2.33.14  2014-09-09
##  biovizBase          1.13.11  2014-09-16
##  bitops              1.0.6    2013-08-17
##  brew                1.0.6    2011-04-13
##  BSgenome            1.33.9   2014-09-12
##  bumphunter          1.5.5    2014-08-29
##  checkmate           1.4      2014-09-03
##  cluster             1.15.3   2014-09-04
##  codetools           0.2.9    2014-08-21
##  colorout          * 1.0.2    2014-03-06
##  colorspace          1.2.4    2013-09-30
##  DBI                 0.3.1    2014-09-24
##  derfinder         * 0.99.2   2014-09-30
##  derfinderHelper     0.99.2   2014-09-29
##  derfinderPlot       0.99.2   2014-09-30
##  devtools          * 1.6      2014-09-23
##  dichromat           2.0.0    2013-01-24
##  digest              0.6.4    2013-12-03
##  doRNG               1.6      2014-03-07
##  evaluate            0.5.5    2014-04-29
##  fail                1.2      2013-09-19
##  foreach             1.4.2    2014-04-11
##  foreign             0.8.61   2014-03-28
##  formatR             1.0      2014-08-25
##  Formula             1.1.2    2014-07-13
##  futile.logger       1.3.7    2014-01-23
##  futile.options      1.0.0    2010-04-06
##  GenomeInfoDb        1.1.20   2014-09-25
##  GenomicAlignments   1.1.30   2014-09-23
##  GenomicFeatures     1.17.15  2014-09-18
##  GenomicFiles        1.1.19   2014-08-26
##  GenomicRanges       1.17.42  2014-09-23
##  GGally              0.4.8    2014-08-26
##  ggbio               1.13.13  2014-09-12
##  ggplot2             1.0.0    2014-05-21
##  graph               1.43.0   2014-04-12
##  gridExtra           0.9.1    2012-08-09
##  gtable              0.1.2    2012-12-05
##  Hmisc               3.14.5   2014-09-12
##  htmltools           0.2.6    2014-09-08
##  httr                0.5      2014-09-02
##  IRanges           * 1.99.28  2014-09-10
##  iterators           1.0.7    2014-04-11
##  knitcitations     * 1.0.2    2014-09-30
##  knitr               1.6      2014-05-25
##  knitrBootstrap    * 1.0.0    2014-09-30
##  lambda.r            1.1.6    2014-01-23
##  lattice             0.20.29  2014-04-04
##  latticeExtra        0.6.26   2013-08-15
##  locfit              1.5.9.1  2013-04-20
##  lubridate           1.3.3    2013-12-31
##  markdown            0.7.4    2014-08-24
##  MASS                7.3.34   2014-08-21
##  Matrix              1.1.4    2014-06-15
##  matrixStats         0.10.0   2014-06-08
##  memoise             0.2.1    2014-04-22
##  mgcv                1.8.3    2014-08-29
##  munsell             0.4.2    2013-07-11
##  nlme                3.1.117  2014-03-31
##  nnet                7.3.8    2014-03-28
##  OrganismDbi         1.7.5    2014-07-31
##  pkgmaker            0.22     2014-05-14
##  plyr                1.8.1    2014-02-26
##  proto               0.3.10   2012-12-22
##  qvalue              1.39.1   2014-05-27
##  R.methodsS3         1.6.1    2014-01-05
##  RBGL                1.41.1   2014-08-14
##  RColorBrewer        1.0.5    2011-06-17
##  Rcpp                0.11.2   2014-06-08
##  RCurl               1.95.4.3 2014-07-29
##  RefManageR          0.8.34   2014-08-19
##  regionReport      * 0.99.3   2014-09-30
##  registry            0.2      2012-01-24
##  reshape             0.8.5    2014-04-23
##  reshape2            1.4      2014-04-23
##  RJSONIO             1.3.0    2014-07-28
##  rmarkdown         * 0.3.3    2014-09-17
##  rngtools            1.2.4    2014-03-06
##  rpart               4.1.8    2014-03-28
##  Rsamtools           1.17.34  2014-09-20
##  RSQLite             0.11.4   2013-05-26
##  rstudioapi          0.1      2014-03-27
##  rtracklayer         1.25.16  2014-09-10
##  S4Vectors         * 0.2.4    2014-09-14
##  scales              0.2.4    2014-04-22
##  sendmailR           1.2.1    2014-09-21
##  stringr             0.6.2    2012-12-06
##  survival            2.37.7   2014-01-22
##  VariantAnnotation   1.11.35  2014-09-18
##  XML                 3.98.1.1 2013-06-20
##  xtable              1.7.4    2014-09-12
##  XVector             0.5.8    2014-09-07
##  yaml                2.1.13   2014-06-12
##  zlibbioc            1.11.1   2014-04-19
##  source                                      
##  CRAN (R 3.1.0)                              
##  Bioconductor                                
##  CRAN (R 3.1.0)                              
##  CRAN (R 3.1.1)                              
##  CRAN (R 3.1.0)                              
##  CRAN (R 3.1.0)                              
##  Bioconductor                                
##  Bioconductor                                
##  Bioconductor                                
##  Bioconductor                                
##  Bioconductor                                
##  Bioconductor                                
##  CRAN (R 3.1.0)                              
##  CRAN (R 3.1.0)                              
##  Bioconductor                                
##  Bioconductor                                
##  CRAN (R 3.1.1)                              
##  CRAN (R 3.1.1)                              
##  CRAN (R 3.1.1)                              
##  local                                       
##  CRAN (R 3.1.0)                              
##  CRAN (R 3.1.1)                              
##  Github (lcolladotor/derfinder@cc7af0a)      
##  Github (lcolladotor/derfinderHelper@54a9721)
##  Github (lcolladotor/derfinderPlot@50e8f51)  
##  CRAN (R 3.1.1)                              
##  CRAN (R 3.1.0)                              
##  CRAN (R 3.1.0)                              
##  CRAN (R 3.1.1)                              
##  CRAN (R 3.1.0)                              
##  CRAN (R 3.1.0)                              
##  CRAN (R 3.1.0)                              
##  CRAN (R 3.1.0)                              
##  CRAN (R 3.1.1)                              
##  CRAN (R 3.1.1)                              
##  CRAN (R 3.1.0)                              
##  CRAN (R 3.1.0)                              
##  Bioconductor                                
##  Bioconductor                                
##  Bioconductor                                
##  Bioconductor                                
##  Bioconductor                                
##  CRAN (R 3.1.1)                              
##  Bioconductor                                
##  CRAN (R 3.1.0)                              
##  Bioconductor                                
##  CRAN (R 3.1.0)                              
##  CRAN (R 3.1.0)                              
##  CRAN (R 3.1.1)                              
##  CRAN (R 3.1.1)                              
##  CRAN (R 3.1.1)                              
##  Bioconductor                                
##  CRAN (R 3.1.0)                              
##  Github (cboettig/knitcitations@f567713)     
##  CRAN (R 3.1.0)                              
##  Github (jimhester/knitrBootstrap@76c41f0)   
##  CRAN (R 3.1.0)                              
##  CRAN (R 3.1.0)                              
##  CRAN (R 3.1.0)                              
##  CRAN (R 3.1.0)                              
##  CRAN (R 3.1.0)                              
##  CRAN (R 3.1.1)                              
##  CRAN (R 3.1.1)                              
##  CRAN (R 3.1.0)                              
##  CRAN (R 3.1.0)                              
##  CRAN (R 3.1.0)                              
##  CRAN (R 3.1.1)                              
##  CRAN (R 3.1.0)                              
##  CRAN (R 3.1.0)                              
##  CRAN (R 3.1.0)                              
##  Bioconductor                                
##  CRAN (R 3.1.0)                              
##  CRAN (R 3.1.0)                              
##  CRAN (R 3.1.0)                              
##  Bioconductor                                
##  CRAN (R 3.1.0)                              
##  Bioconductor                                
##  CRAN (R 3.1.0)                              
##  CRAN (R 3.1.0)                              
##  CRAN (R 3.1.1)                              
##  CRAN (R 3.1.1)                              
##  Github (lcolladotor/regionReport@c980423)   
##  CRAN (R 3.1.0)                              
##  CRAN (R 3.1.0)                              
##  CRAN (R 3.1.0)                              
##  CRAN (R 3.1.1)                              
##  CRAN (R 3.1.1)                              
##  CRAN (R 3.1.0)                              
##  CRAN (R 3.1.0)                              
##  Bioconductor                                
##  CRAN (R 3.1.0)                              
##  CRAN (R 3.1.0)                              
##  Bioconductor                                
##  Bioconductor                                
##  CRAN (R 3.1.0)                              
##  CRAN (R 3.1.1)                              
##  CRAN (R 3.1.0)                              
##  CRAN (R 3.1.0)                              
##  Bioconductor                                
##  CRAN (R 3.1.0)                              
##  CRAN (R 3.1.1)                              
##  Bioconductor                                
##  CRAN (R 3.1.0)                              
##  Bioconductor
</code></pre>
</div>
<h1>Bibliography</h1>
<p>This vignette was generated using <code>knitrBootstrap</code> (<a href='https://github.com/jimhester/'>Hester, 2014</a>) with <code>knitr</code> (<a href='http://www.crcpress.com/product/isbn/9781466561595'>Xie, 2014</a>) and <code>rmarkdown</code> (<a href='http://CRAN.R-project.org/package=rmarkdown'>Allaire, McPherson, Xie, Wickham, et al., 2014</a>) running behind the scenes.</p>
<p>Citations made with <code>knitcitations</code> (<a href='https://github.com/cboettig/knitcitations'>Boettiger, 2014</a>).</p>
<div class="row">
<p>
<a id='bib-rmarkdown'></a><a href="#cite-rmarkdown">[1]</a><cite> J. Allaire, J. McPherson, Y. Xie, H. Wickham, et al. <EM>rmarkdown: Dynamic Documents for R</EM>. R package version 0.3.3. 2014. URL: <a href="http://CRAN.R-project.org/package=rmarkdown">http://CRAN.R-project.org/package=rmarkdown</a>.</cite>
</p>
<p>
<a id='bib-GenomeInfoDb'></a><a href="#cite-GenomeInfoDb">[2]</a><cite> S. Arora, M. Morgan, M. Carlson and H. Pages. <EM>GenomeInfoDb: Utilities for manipulating chromosome and other 'seqname' identifiers</EM>. R package version 1.1.20. 2014.</cite>
</p>
<p>
<a id='bib-gridExtra'></a><a href="#cite-gridExtra">[3]</a><cite> B. Auguie. <EM>gridExtra: functions in Grid graphics</EM>. R package version 0.9.1. 2012. URL: <a href="http://CRAN.R-project.org/package=gridExtra">http://CRAN.R-project.org/package=gridExtra</a>.</cite>
</p>
<p>
<a id='bib-knitcitations'></a><a href="#cite-knitcitations">[4]</a><cite> C. Boettiger. <EM>knitcitations: Citations for knitr markdown files</EM>. R package version 1.0.2. 2014. URL: <a href="https://github.com/cboettig/knitcitations">https://github.com/cboettig/knitcitations</a>.</cite>
</p>
<p>
<a id='bib-TxDbHsapiensUCSChg19knownGene'></a><a href="#cite-TxDbHsapiensUCSChg19knownGene">[5]</a><cite> M. Carlson. <EM>TxDb.Hsapiens.UCSC.hg19.knownGene: Annotation package for TranscriptDb object(s)</EM>. R package version 2.14.0. 2014.</cite>
</p>
<p>
<a id='bib-derfinder'></a><a href="#cite-derfinder">[6]</a><cite> L. Collado-Torres, A. Frazee, A. Jaffe and J. Leek. <EM>derfinder: Fast differential expression analysis of RNA-seq data at base-pair resolution</EM>. R package version 0.99.2. 2014. URL: <a href="https://github.com/lcolladotor/derfinder">https://github.com/lcolladotor/derfinder</a>.</cite>
</p>
<p>
<a id='bib-derfinderPlot'></a><a href="#cite-derfinderPlot">[7]</a><cite> L. Collado-Torres, A. Jaffe and J. Leek. <EM>derfinderPlot: Plotting functions for derfinder</EM>. R package version 0.99.2. 2014. URL: <a href="https://github.com/lcolladotor/derfinderPlot">https://github.com/lcolladotor/derfinderPlot</a>.</cite>
</p>
<p>
<a id='bib-regionReport'></a><a href="#cite-regionReport">[8]</a><cite> L. Collado-Torres, A. Jaffe and J. Leek. <EM>regionReport: Generate HTML reports for exploring a set of regions</EM>. R package version 0.99.3. 2014. URL: <a href="https://github.com/lcolladotor/regionReport">https://github.com/lcolladotor/regionReport</a>.</cite>
</p>
<p>
<a id='bib-knitrBootstrap'></a><a href="#cite-knitrBootstrap">[9]</a><cite> J. Hester. <EM>knitrBootstrap: Knitr Bootstrap framework.</EM> R package version 1.0.0. 2014. URL: <a href="https://github.com/jimhester/">https://github.com/jimhester/</a>.</cite>
</p>
<p>
<a id='bib-GenomicRanges'></a><a href="#cite-GenomicRanges">[10]</a><cite> M. Lawrence, W. Huber, H. Pagès, P. Aboyoun, et al. “Software for Computing and Annotating Genomic Ranges”. In: <EM>PLoS Computational Biology</EM> 9 (8 2013). DOI: <a href="http://dx.doi.org/10.1371/journal.pcbi.1003118">10.1371/journal.pcbi.1003118</a>. URL: <a href="http://www.ploscompbiol.org/article/info%3Adoi%2F10.1371%2Fjournal.pcbi.1003118}.">http://www.ploscompbiol.org/article/info%3Adoi%2F10.1371%2Fjournal.pcbi.1003118}.</a>
</p>
</cite>
</p>
<p>
<a id='bib-RColorBrewer'></a><a href="#cite-RColorBrewer">[11]</a><cite> E. Neuwirth. <EM>RColorBrewer: ColorBrewer palettes</EM>. R package version 1.0-5. 2011. URL: <a href="http://CRAN.R-project.org/package=RColorBrewer">http://CRAN.R-project.org/package=RColorBrewer</a>.</cite>
</p>
<p>
<a id='bib-R'></a><a href="#cite-R">[12]</a><cite> R Core Team. <EM>R: A Language and Environment for Statistical Computing</EM>. R Foundation for Statistical Computing. Vienna, Austria, 2014. URL: <a href="http://www.R-project.org/">http://www.R-project.org/</a>.</cite>
</p>
<p>
<a id='bib-Cairo'></a><a href="#cite-Cairo">[13]</a><cite> S. Urbanek and J. Horner. <EM>Cairo: R graphics device using cairo graphics library for creating high-quality bitmap (PNG, JPEG, TIFF), vector (PDF, SVG, PostScript) and display (X11 and Win32) output.</EM> R package version 1.5-5. 2013. URL: <a href="http://CRAN.R-project.org/package=Cairo">http://CRAN.R-project.org/package=Cairo</a>.</cite>
</p>
<p>
<a id='bib-ggplot2'></a><a href="#cite-ggplot2">[14]</a><cite> H. Wickham. <EM>ggplot2: elegant graphics for data analysis</EM>. Springer New York, 2009. ISBN: 978-0-387-98140-6. URL: <a href="http://had.co.nz/ggplot2/book">http://had.co.nz/ggplot2/book</a>.</cite>
</p>
<p>
<a id='bib-devtools'></a><a href="#cite-devtools">[15]</a><cite> H. Wickham and W. Chang. <EM>devtools: Tools to make developing R code easier</EM>. R package version 1.6. 2014. URL: <a href="http://CRAN.R-project.org/package=devtools">http://CRAN.R-project.org/package=devtools</a>.</cite>
</p>
<p>
<a id='bib-knitr'></a><a href="#cite-knitr">[16]</a><cite> Y. Xie. “knitr: A Comprehensive Tool for Reproducible Research in R”. In: <EM>Implementing Reproducible Computational Research</EM>. Ed. by V. Stodden, F. Leisch and R. D. Peng. ISBN 978-1466561595. Chapman and Hall/CRC, 2014. URL: <a href="http://www.crcpress.com/product/isbn/9781466561595">http://www.crcpress.com/product/isbn/9781466561595</a>.</cite>
</p>
<p>
<a id='bib-ggbio'></a><a href="#cite-ggbio">[17]</a><cite> T. Yin, D. Cook and M. Lawrence. “ggbio: an R package for extending the grammar of graphics for genomic data”. In: <EM>Genome Biology</EM> 13.8 (2012), p. R77.</cite>
</p>
<p>
<a id='bib-biovizBase'></a><a href="#cite-biovizBase">[18]</a><cite> T. Yin, M. Lawrence and D. Cook. <EM>biovizBase: Basic graphic utilities for visualization of genomic data.</EM> R package version 1.13.11. 2014.</cite>
</p>
</div>
</div>
</div>
<div class="navbar navbar-fixed-bottom navbar-inverse">
<div class="container">
<div class="navbar-header">
<button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".navbar-responsive-collapse">
<span class="icon-bar"></span> <span class="icon-bar"></span> <span class="icon-bar"></span>
</button>
</div>
<div id="bottom-navbar" class="navbar-collapse collapse navbar-responsive-collapse">
<ul class="nav navbar-nav navbar-right">
<li class="nav">
<p class="navbar-text">
Toggle
</p>
</li>
<li class="dropup">
<a href="#" class="dropdown-toggle" data-toggle="dropdown">Code <b class="caret"></b></a>
<ul class="dropdown-menu">
<li class="dropdown-header">
Languages
</li>
<li class="active">
<a href="#" class="toggle-global source R" type="source.R">R</a>
</li>
<li>
<a href="#" type="all-source" class="toggle-global">All</a>
</li>
</ul>
</li>
<li class="dropup">
<a href="#" class="dropdown-toggle" data-toggle="dropdown">Output <b class="caret"></b></a>
<ul class="dropdown-menu">
<li class="dropdown-header">
Type
</li>
<li class="active">
<a href="#" class="toggle-global output" type="output">output</a>
</li>
<li class="active">
<a href="#" class="toggle-global message" type="message">message</a>
</li>
<li>
<a href="#" type="all-output" class="toggle-global">All</a>
</li>
</ul>
</li>
<li class="active">
<a href="#" type="figure" class="toggle-global">Figures</a>
</li>
</ul>
<ul class="nav navbar-nav navbar-left">
<li class="dropup">
<a href="#" class="dropdown-toggle" data-toggle="dropdown">Bootstrap theme <b class="caret"></b></a>
<ul class="dropdown-menu">
<li class="active">
<a class="theme-switch" title="Default" href="#">Default</a>
</li>
<li>
<a class="theme-switch" title="Amelia" href="#">Amelia</a>
</li>
<li>
<a class="theme-switch" title="Cerulean" href="#">Cerulean</a>
</li>
<li>
<a class="theme-switch" title="Cosmo" href="#">Cosmo</a>
</li>
<li>
<a class="theme-switch" title="Cyborg" href="#">Cyborg</a>
</li>
<li>
<a class="theme-switch" title="Journal" href="#">Journal</a>
</li>
<li>
<a class="theme-switch" title="Flatly" href="#">Flatly</a>
</li>
<li>
<a class="theme-switch" title="Readable" href="#">Readable</a>
</li>
<li>
<a class="theme-switch" title="Simplex" href="#">Simplex</a>
</li>
<li>
<a class="theme-switch" title="Slate" href="#">Slate</a>
</li>
<li>
<a class="theme-switch" title="Spacelab" href="#">Spacelab</a>
</li>
<li>
<a class="theme-switch" title="United" href="#">United</a>
</li>
</ul>
</li>
</ul>
<ul class="nav navbar-nav navbar-left">
<li class="dropup">
<a href="#" class="dropdown-toggle" data-toggle="dropdown">Code highlighting <b class="caret"></b></a>
<ul class="dropdown-menu">
<li class="active">
<a class="highlight-switch" title="HighlightJs" href="#">HighlightJs</a>
</li>
<li>
<a class="highlight-switch" title="Dark" href="#">Dark</a>
</li>
<li>
<a class="highlight-switch" title="Far" href="#">Far</a>
</li>
<li>
<a class="highlight-switch" title="Idea" href="#">Idea</a>
</li>
<li>
<a class="highlight-switch" title="Sunburst" href="#">Sunburst</a>
</li>
<li>
<a class="highlight-switch" title="Zenburn" href="#">Zenburn</a>
</li>
<li>
<a class="highlight-switch" title="Visual Studio" href="#">Visual Studio</a>
</li>
<li>
<a class="highlight-switch" title="Ascetic" href="#">Ascetic</a>
</li>
<li>
<a class="highlight-switch" title="Magula" href="#">Magula</a>
</li>
<li>
<a class="highlight-switch" title="Github" href="#">Github</a>
</li>
<li>
<a class="highlight-switch" title="Google Code" href="#">Google Code</a>
</li>
<li>
<a class="highlight-switch" title="Brown Paper" href="#">Brown Paper</a>
</li>
<li>
<a class="highlight-switch" title="School Book" href="#">School Book</a>
</li>
<li>
<a class="highlight-switch" title="IR Black" href="#">IR Black</a>
</li>
<li>
<a class="highlight-switch" title="Solarized - Dark" href="#">Solarized - Dark</a>
</li>
<li>
<a class="highlight-switch" title="Solarized - Light" href="#">Solarized - Light</a>
</li>
<li>
<a class="highlight-switch" title="Arta" href="#">Arta</a>
</li>
<li>
<a class="highlight-switch" title="Monokai" href="#">Monokai</a>
</li>
<li>
<a class="highlight-switch" title="Xcode" href="#">Xcode</a>
</li>
<li>
<a class="highlight-switch" title="Pojoaque" href="#">Pojoaque</a>
</li>
<li>
<a class="highlight-switch" title="Rainbow" href="#">Rainbow</a>
</li>
<li>
<a class="highlight-switch" title="Tomorrow" href="#">Tomorrow</a>
</li>
<li>
<a class="highlight-switch" title="Tomorrow Night" href="#">Tomorrow Night</a>
</li>
<li>
<a class="highlight-switch" title="Tomorrow Night Bright" href="#">Tomorrow Night Bright</a>
</li>
<li>
<a class="highlight-switch" title="Tomorrow Night Blue" href="#">Tomorrow Night Blue</a>
</li>
<li>
<a class="highlight-switch" title="Tomorrow Night Eighties" href="#">Tomorrow Night Eighties</a>
</li>
</ul>
</li>
</ul>
</div>
</div>
</div>
<link rel="alternate stylesheet" type="text/css" href="https://netdna.bootstrapcdn.com/bootstrap/3.0.0/css/bootstrap.min.css" title="Default" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://netdna.bootstrapcdn.com/bootswatch/3.0.0/amelia/bootstrap.min.css" title="Amelia" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://netdna.bootstrapcdn.com/bootswatch/3.0.0/cerulean/bootstrap.min.css" title="Cerulean" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://netdna.bootstrapcdn.com/bootswatch/3.0.0/cosmo/bootstrap.min.css" title="Cosmo" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://netdna.bootstrapcdn.com/bootswatch/3.0.0/cyborg/bootstrap.min.css" title="Cyborg" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://netdna.bootstrapcdn.com/bootswatch/3.0.0/journal/bootstrap.min.css" title="Journal" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://netdna.bootstrapcdn.com/bootswatch/3.0.0/flatly/bootstrap.min.css" title="Flatly" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://netdna.bootstrapcdn.com/bootswatch/3.0.0/readable/bootstrap.min.css" title="Readable" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://netdna.bootstrapcdn.com/bootswatch/3.0.0/simplex/bootstrap.min.css" title="Simplex" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://netdna.bootstrapcdn.com/bootswatch/3.0.0/slate/bootstrap.min.css" title="Slate" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://netdna.bootstrapcdn.com/bootswatch/3.0.0/spacelab/bootstrap.min.css" title="Spacelab" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://netdna.bootstrapcdn.com/bootswatch/3.0.0/united/bootstrap.min.css" title="United" media="screen"></link><link rel="alternate stylesheet" type="text/css" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/7.3/styles/default.min.css" title="HighlightJs" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/7.3/styles/dark.min.css" title="Dark" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/7.3/styles/far.min.css" title="Far" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/7.3/styles/idea.min.css" title="Idea" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/7.3/styles/sunburst.min.css" title="Sunburst" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/7.3/styles/zenburn.min.css" title="Zenburn" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/7.3/styles/vs.min.css" title="Visual Studio" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/7.3/styles/ascetic.min.css" title="Ascetic" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/7.3/styles/magula.min.css" title="Magula" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/7.3/styles/github.min.css" title="Github" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/7.3/styles/googlecode.min.css" title="Google Code" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/7.3/styles/brown_paper.min.css" title="Brown Paper" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/7.3/styles/school_book.min.css" title="School Book" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/7.3/styles/ir_black.min.css" title="IR Black" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/7.3/styles/solarized_dark.min.css" title="Solarized - Dark" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/7.3/styles/solarized_light.min.css" title="Solarized - Light" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/7.3/styles/arta.min.css" title="Arta" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/7.3/styles/monokai.min.css" title="Monokai" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/7.3/styles/xcode.min.css" title="Xcode" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/7.3/styles/pojoaque.min.css" title="Pojoaque" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/7.3/styles/rainbow.min.css" title="Rainbow" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/7.3/styles/tomorrow.min.css" title="Tomorrow" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/7.3/styles/tomorrow-night.min.css" title="Tomorrow Night" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/7.3/styles/tomorrow-night-bright.min.css" title="Tomorrow Night Bright" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/7.3/styles/tomorrow-night-blue.min.css" title="Tomorrow Night Blue" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/7.3/styles/tomorrow-night-eighties.min.css" title="Tomorrow Night Eighties" media="screen"></link>
</div>
<div id="push">

</div>
<div id="footer">
<div class="container">
<p class="text-muted" id="credit">
Styled with <a href="https://github.com/jimhester/knitrBootstrap">knitrBootstrap</a>
</p>
</div>
</div>
<link rel="stylesheet" id="theme" href="https://netdna.bootstrapcdn.com/bootstrap/3.0.0/css/bootstrap.min.css" media="screen"></link><link rel="stylesheet" id="highlight" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/7.3/styles/default.min.css" media="screen"></link>
</div>
</body>
</html>
